### PR TITLE
Reusable workflows for eden tests

### PR DESCRIPTION
**Pull request by yash-zededa**
This PR will allow the [lf-edge/eve](https://github.com/lf-edge/eve/actions) to call the workflow from eden repo i.e re-usable workflow. 

The configurations would be DRY and all the changes in eden can then be managed without having to copy them over to the eve.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -25,7 +30,7 @@ jobs:
   networking: 
     name: Networking test suite
     needs: smoke
-    runs-on: ubuntu-22.04
+    runs-on: self-hosted
```
Comment (uncleDecart) [inline .github/workflows/test.yml:33]: I assume this is temp changes

File: .github/actions/setup-environment/action.yml [hunk 2]
```diff
@@ -23,9 +23,9 @@ runs:
               echo "$addr overlaps with test"; exit 1
             fi
         done
-        sudo df -h
+        df -h
```
[the remark this answers is not in the document]
Comment (yash-zededa) [reply .github/actions/setup-environment/action.yml:26]: We don't need sudo for `df -h`, right ?

File: .github/workflows/test.yml (hunk 1 above)
Comment (yash-zededa) [reply .github/workflows/test.yml:33]: Yes, these are temp changes

File: .github/actions/setup-environment/action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (uncleDecart) [reply .github/actions/setup-environment/action.yml:26]: no, we don't you're correct

File: .github/workflows/test.yml (hunk 1 above)
Comment (yash-zededa) [reply .github/workflows/test.yml]: Temp changes reverted.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -61,8 +76,12 @@ jobs:
     steps:
       - name: Get code
         uses: actions/checkout@v3
+        with:
+          repository: yash-zededa/eden
```
Comment (uncleDecart) [inline .github/workflows/test.yml:80]: Maybe you should remove your fork from here?

Comment (yash-zededa) [reply .github/workflows/test.yml:80]: I have used fork to test the actions from eden. It works now so we are good to point to lf-edge/eden

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -79,8 +98,12 @@ jobs:
...
```
Comment (uncleDecart) [inline .github/workflows/test.yml:102]: Here as well you're referring to your fork

File: .github/actions/publish-logs/action.yml [hunk 5]
```diff
@@ -22,6 +22,7 @@ runs:
         cp dist/default-eve.log console.log || echo "no device log"
         docker logs eden_adam > adam.log 2>&1 || echo "no adam log"
       shell: bash
+      working-directory: ./eden_repo
```
[the remark this answers is not in the document]
Comment (uncleDecart) [reply .github/actions/publish-logs/action.yml]: You're right, let's do it like this

File: .github/workflows/test.yml (hunk 4 above)
Comment (yash-zededa) [reply .github/workflows/test.yml:102]: Yes, I have repointed to my fork to test the workflow by manual trigger from eden.  It works now so we are good to point to lf-edge/eden.

**Comment by uncleDecart** [comment]
LGTM, some minor comments with your fork usage and path input variable

**Comment by milan-zededa** [comment]
Do I get it right that this will allow us to get rid of the eden copy in the eve repo, i.e [this directory](https://github.com/lf-edge/eve/tree/master/tests/eden)?

File: .github/actions/setup-environment/action.yml [hunk 6]
```diff
@@ -36,21 +36,25 @@ runs:
       run: |
         make build-tests
       shell: bash
+      working-directory: "./eden_repo"
```
Comment (uncleDecart) [inline .github/actions/setup-environment/action.yml:39]: Do we need to provide `working-directory` for every step in every custom action or will it be overridden by high level `working-directory` parameter i.e. in `test.yml` workflow?

Comment (yash-zededa) [reply .github/actions/setup-environment/action.yml:39]: It's a good suggestion. Let me try. As per github docs, The `working-directory` attribute will be set only for the `run` section so it should not conflict with the `path`  attribute of the `uses` checkout action

**Review by uncleDecart — APPROVED** [review]
LGTM, one small comment about `working-directory`

File: .github/actions/setup-environment/action.yml (hunk 6 above)
Comment (yash-zededa) [reply .github/actions/setup-environment/action.yml:39]: I am trying to fix it at the test.yml workflow level :). Let's see if it works. If not then defaults at the action level

**Comment by uncleDecart** [comment]
Hey @yash-zededa, could you rebase to master, looks like there're some small merge conflict.

**Comment by yash-zededa** [comment]
> Hey @yash-zededa, could you rebase to master, looks like there're some small merge conflict.

Sure,I will rebase it, I am testing the `working-directory` changed with the test.yml

File: .github/actions/setup-environment/action.yml (hunk 6 above)
Comment (yash-zededa) [reply .github/actions/setup-environment/action.yml:40]: The `defaults` sections is not supported in actions. Need to set the `shell: bash` and `working-directory: "./eden_repo"` for all the `run` commands explicitly.

Comment (uncleDecart) [reply .github/actions/setup-environment/action.yml:40]: Well, at least we tried :D

**Pull request closed without merging**